### PR TITLE
fix(transition-ios): RTL fix for transition on ios

### PR DESCRIPTION
**Pull request by sijav**
#### Short description of what this resolves:
This will fix the transition on ios platform when it's RTL

#### Changes proposed in this pull request:

- Changed the `translateX` animations with opposite value if it's on rtl

**Ionic Version**: 2.x / 3.x

**Fixes**: #11211

### CODE REVIEW

**Comment by AmitMY** [comment]
Works very well!
I am not very sure about the naming though, as `OFF_LEFT_RTL` seems to really mean the offset to the right.

Do you think using `start` and `end` makes sense for this? like `OFF_START`, `OFF_END`, `OFF_START_RTL`, `OFF_END_RTL`? (left is start, right is end)

**Comment by sijav** [comment]
Totally agree, Start and End should be better for this, although I have a better solution not to add any variable but to add a '-' at behind of the affected transition (as it is a string value)
I'll make another commit and let me know which one is better.

**Comment by sijav** [comment]
@AmitMY nevermind what I was saying earlier :D it was the worst idea, besides it really is doing it really is right offset not start offset, but we do it -99.5% instead of 99.5%

**Comment by AmitMY** [comment]
@sijav Yeah I saw.. I was just about to comment the `--` case, but you reverted :)
Keeping it right & left seems reasonable (as two separate cases).
There is a solution to do with only one set of variables, but I am not sure the overhead is worth it.
(have a utility function that gets a number/string, casts it into a string, then checks if the first character is `-`, and if so removes it, otherwise adds it)

**Comment by sijav** [comment]
@AmitMY yeah ... it was a very bad solution :D
I don't think it worth it but let me know if anything could be done...

**Comment by brandyscarney** [comment]
Thanks for the PR! This looks good to me. I want to get one more team member to test it and if it looks good to them we'll merge it. :smile:

File: src/transitions/transition-ios.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ const TRANSFORM = 'transform';
 const TRANSLATEX = 'translateX';
 const OFF_RIGHT = '99.5%';
 const OFF_LEFT = '-33%';
+const OFF_RIGHT_RTL = '-99.5%';
+const OFF_LEFT_RTL = '33%';
```
Comment (Khalid-Nowaf) [inline src/transitions/transition-ios.ts]: what do think if you use `OFF_RIGHT` --> `OFF_START`, `OFF_LEFT` --> `OFF_END`  ? 
then you use the check `_plt.isRTL` just one time. Also you would  help the framwork to adapt `Start` and `End` notations 
````js
OFF_START = _plt.isRTL ? '-99.5%' : '99.5%';
OFF_END = _plt.isRTL ? '33%' : '-33%';
`````

**Comment by AmitMY** [comment]
Khalid had a good point. Instead of duplicating the variables, you can do:
OFF_LEFT = this.plt.isRTL ? "-33%" : "33%"

So you only check once (per call) and it is cleaner.

What do u think?

**Comment by sijav** [comment]
@AmitMY Khalid? I don't see any comment from him, but nevertheless I have nothing against it however I don't know how RTL change can work in runtime but I hope this change won't affect it.

**Comment by AmitMY** [comment]
@sijav Well, of course that part of it will be to move the constants to the init function, which I think will mean that change will work (if it works for current code)

Khalid commented but removed his comment.

**Comment by Khalid-Nowaf** [comment]
Sorry guys, I just removed my comment after seen my suggestion is already suggested by @AmitMY.

**Comment by sijav** [comment]
@AmitMY How can I do that?
The `OFF_LEFT` is a const value we need to declare it's value only when we declare the variable, and where we declare it is on the root of the ts not on any class, and it will be called at the init of the program, so there's no `this.plt` available.

File: src/transitions/transition-ios.ts [hunk 2]
```diff
@@ -20,6 +18,8 @@ export class IOSTransition extends PageTransition {
     super.init();
 
     const plt = this.plt;
+    const OFF_RIGHT = plt.isRTL ? '-99.5%' : '99.5%';
+    const OFF_LEFT = plt.isRTL ? '33%' : '-33%';
```
Comment (AmitMY) [inline src/transitions/transition-ios.ts]: Looks much better! @brandyscarney any other note or can this get merged?

Comment (brandyscarney) [inline src/transitions/transition-ios.ts]: Nice! I have someone else testing it today and if all goes well with their tests I will merge it. 🙂

**Comment by mhartington** [comment]
Only thing I can see is that the swipe to go back gesture isn't moving the active page, but that issue existed before this PR, so I think it's ok for now.

**Comment by sijav** [comment]
@mhartington I've made about a dozen PR #11822 will fix that

**Comment by brandyscarney** [comment]
Perfect, thanks!

**Pull request merged**